### PR TITLE
GB3-1724:  hamburger menu

### PR DESCRIPTION
**Pull request by TIL-EBP**
Adds hamburger Menu for meduium Breakpoint.
Adds link to profile page

### CODE REVIEW

File: src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html [hunk 1]
```diff
@@ -1,13 +1,21 @@
 <mat-card class="navbar-mobile__dialog">
   <mat-card-header class="navbar-mobile__dialog__header">
-    <a (click)="this.startLogin()" *ngIf="!this.isAuthenticated; else userProfile" class="navbar__item" mat-button>Login</a>
+    <a (click)="this.startLogin()" *ngIf="!this.isAuthenticated; else userProfile" class="navbar__item navbar__user" mat-button>
```
Comment (wendlans) [inline src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html:3]: Quick question: do we replace `ngIf` now with `@if` as soon as we change a template that contains them? Or are we leaving them as they are?

File: src/app/shared/components/navbar/navbar.component.ts [hunk 2]
```diff
@@ -28,7 +31,10 @@ export class NavbarComponent implements OnInit, OnDestroy {
   private readonly screenMode$ = this.store.select(selectScreenMode);
   private readonly isAuthenticated$ = this.store.select(selectIsAuthenticated);
 
-  constructor(private readonly store: Store) {}
+  constructor(
+    private readonly store: Store,
+    public dialog: MatDialog,
```
Comment (wendlans) [inline src/app/shared/components/navbar/navbar.component.ts:36]:
```suggestion
    private readonly dialog: MatDialog,
```

File: src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
+      <mat-icon svgIcon="ktzh_user"></mat-icon>
+      <span>Login</span>
```
Comment (wendlans) [inline src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html:5]: Also: I've quickly tested this by navigating only using the keyboard. I'm not able to reach the `login` or the logged in user using tab or the arrow keys. Could also be _skill issues_ on my part 😅

File: src/app/shared/components/navbar/navbar.component.scss [hunk 4]
```diff
@@ -5,12 +5,22 @@
...
   position: relative;
   height: 100%;
-  font-size: 14px;
+  font-size: 16px;
 
   .navbar__spacer {
     flex: 1 1 auto;
   }
 
+  .navbar__menu__text {
```
Comment (wendlans) [inline src/app/shared/components/navbar/navbar.component.scss:14]: `.navbar__menu` does not exist but is used within the template. Either delete `.navbar__menu` in the template and rename this one to `.navbar__menu-text` or create `.navbar__menu`.

File: src/app/shared/components/navbar/navbar.component.html [hunk 5]
```diff
@@ -12,57 +12,74 @@
...
+    @if (!this.isAuthenticated) {
+      <button (click)="this.startLogin()" class="navbar__item navbar__item--button" mat-button>
+        <mat-icon svgIcon="ktzh_user"></mat-icon>Login
       </button>
     } @else {
-      <button [matMenuTriggerFor]="userMenu" class="navbar__item navbar__item--button" mat-button #trigger="matMenuTrigger">
+      <button [matMenuTriggerFor]="userMenu" class="navbar__item navbar__item--button" mat-button>
         <mat-icon svgIcon="ktzh_user"></mat-icon>
-        <mat-icon [fontIcon]="trigger.menuOpen ? 'arrow_drop_up' : 'arrow_drop_down'"></mat-icon>
+        {{ this.userName }}
```
Comment (wendlans) [inline src/app/shared/components/navbar/navbar.component.html:62]: unnecessary `this`

**Edit:** there are a couple more unnecessary `this` within this template

**Review by wendlans — APPROVED** [review]
Good PR I've just found some minor code issues and an accessibility challenge.

File: src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html (hunk 1 above)
Comment (TIL-EBP) [reply src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html:3]: Good Question. I would say yes

File: src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html (hunk 3 above)
Comment (TIL-EBP) [reply src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html:5]: True, I changed it to be a button (same as on desktop nav), so it is no reachable via tab

File: src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html (hunk 1 above)
Comment (wendlans) [reply src/app/shared/components/navbar-mobile/navbar-mobile-dialog/navbar-mobile-dialog.component.html]: > _we change a template_ (...) _Or are we leaving them_

**Yes**
😂😂😂

But thanks for upgrading them to `@if/@else` 👍 I would have suggested this approach as well.

**Review by wendlans — APPROVED** [review]
Looks all good now 👌

**Pull request merged**